src/CMakeLists.txt

```diff
@@ -26,34 +26,72 @@ set(SOURCES
+set(PKG_CONFIG_FILE     "${PKG_DIR}/${TARGET_NAME}.pc")
+set(CONFIG_FILE         "${PKG_DIR}/${PROJECT_NAME}Config.cmake")
+set(VERSION_FILE        "${PKG_DIR}/${PROJECT_NAME}ConfigVersion.cmake")
+set(FIND_RDKAFKA_FILE   "${PROJECT_SOURCE_DIR}/cmake/FindRdKafka.cmake")
+set(NAMESPACE           "${PROJECT_NAME}::")
+set(TARGET_EXPORT_NAME  ${PROJECT_NAME}Targets)
 
-add_library(cppkafka ${CPPKAFKA_LIBRARY_TYPE} ${SOURCES})
-set_target_properties(cppkafka PROPERTIES VERSION ${CPPKAFKA_VERSION}
-                                          SOVERSION ${CPPKAFKA_VERSION})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../include/cppkafka)
+include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 
-set(DEPENDENCIES ${RDKAFKA_LIBRARY})
+add_library(${TARGET_NAME} ${CPPKAFKA_LIBRARY_TYPE} ${SOURCES})
+set_target_properties(${TARGET_NAME} PROPERTIES VERSION ${CPPKAFKA_VERSION}
+                                                 SOVERSION ${CPPKAFKA_VERSION})
+# In CMake >= 3.15 Boost::boost == Boost::headers
+set(DEPENDENCIES ${RdKafka_DEPENDENCIES})
 if (WIN32)
     # On windows ntohs and related are in ws2_32
     set(DEPENDENCIES ${DEPENDENCIES} ws2_32.lib)
 endif()
-target_link_libraries(cppkafka ${DEPENDENCIES})
-target_include_directories(cppkafka PUBLIC ${PROJECT_SOURCE_DIR}/include)
+target_link_libraries(${TARGET_NAME} PRIVATE ${DEPENDENCIES})
 
+# Install cppkafka target and specify all properties needed for the exported file
 install(
-    TARGETS cppkafka
-    LIBRARY DESTINATION ${LIBDIR}
-    ARCHIVE DESTINATION ${LIBDIR}
+    TARGETS ${TARGET_NAME}
+    EXPORT ${TARGET_EXPORT_NAME}
+    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+    INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
     COMPONENT dev
 )
 
-# Generate package configuration file
-set(PKG_CONFIG ${PROJECT_SOURCE_DIR}/package/cppkafka.pc)
-configure_file(${PROJECT_SOURCE_DIR}/cppkafka.pc.in ${PKG_CONFIG} @ONLY)
+# Install the exported file
+install(
+    EXPORT "${TARGET_EXPORT_NAME}"
+    NAMESPACE "${NAMESPACE}"
+    DESTINATION "${CPPKAFKA_CONFIG_DIR}"
+)
+
+# Generate and install pkgconfig file
+configure_file(${PROJECT_SOURCE_DIR}/cmake/cppkafka.pc.in ${PKG_CONFIG_FILE} @ONLY)
 
 install(
     FILES ${PKG_CONFIG}
-    DESTINATION ${CPPKAFKA_PKGCONFIG_DIR}
+    DESTINATION "${CPPKAFKA_PKGCONFIG_DIR}"
     COMPONENT pkgconfig
 )
+
+# Generate CMAKE configuration file and exported targets
+configure_package_config_file(
+    "${PROJECT_SOURCE_DIR}/cmake/config.cmake.in"
+    "${CONFIG_FILE}"
+    INSTALL_DESTINATION "${CPPKAFKA_CONFIG_DIR}"
+    PATH_VARS CMAKE_INSTALL_PREFIX CMAKE_INSTALL_INCLUDEDIR CMAKE_INSTALL_LIBDIR
+)
+
+# Generate version file
+write_basic_package_version_file(
+    "${VERSION_FILE}"
+    VERSION ${CPPKAFKA_VERSION}
+    COMPATIBILITY AnyNewerVersion
+)
+
+install(
+    FILES "${CONFIG_FILE}" "${VERSION_FILE}" "${FIND_RDKAFKA_FILE}"
```

Comment:
Is exporting the rdkafka find file going to cause issues for users who have built rdkafka using cmake and already have a find script?

Reply:
It should not. The CppKafkaConfig.cmake will first search for the RdKafkaConfig.cmake which will be found if they have built RdKafka with cmake, so in this case your find script will not be invoked.